File: zsbApp.PreviewPdfByFoxit/FoxitPDFViewerForNET200_32914_enu/Examples/Visual Studio 2010_2012_2013_2015/VB/CustomSearchDialog/Form1.vb
Public Class Form1

    Public Sub New()
        MyBase.New()
        InitializeComponent()
        ' In the form constructor do not use any Viewer methods (Open, Print etc.) that perform UI activities.
    End Sub
    Private Sub pdfViewer1_SearchDialogShowing(ByVal sender As System.Object, ByVal e As Foxit.PDF.Viewer.SearchDialogShowingEventArgs)
        e.Cancel = True
        searchText.Focus()
    End Sub
    Private Sub button3_Click(ByVal sender As System.Object, ByVal e As EventArgs)
        Dim matchOptions As Foxit.PDF.Viewer.MatchOptions
        If (Not String.IsNullOrEmpty(searchText.Text)) Then
            matchOptions = Foxit.PDF.Viewer.MatchOptions.All
            If ((matchWholeWord.Checked = True) And (matchCase.Checked = True)) Then
                matchOptions = Foxit.PDF.Viewer.MatchOptions.All
            ElseIf ((matchWholeWord.Checked = True) And (matchCase.Checked = False)) Then
                matchOptions = Foxit.PDF.Viewer.MatchOptions.WholeWordOnly
            ElseIf ((matchWholeWord.Checked = False) And (matchCase.Checked = True)) Then
                matchOptions = Foxit.PDF.Viewer.MatchOptions.CaseSensitive
            Else
                matchOptions = Foxit.PDF.Viewer.MatchOptions.None
            End If
            pdfViewer1.SearchForward(searchText.Text, matchOptions)
        End If
    End Sub
    Private Sub pdfViewer1_DocumentDisplayed(ByVal sender As System.Object, ByVal e As Foxit.PDF.Viewer.DocumentDisplayedEventArgs)
        groupBox1.Enabled = True
        title.Text = "Type a search term in the 'Find' text box and click on 'Find Next'."
    End Sub
End Class

File: zsbApp.PreviewPdfByFoxit/FoxitPDFViewerForNET200_32914_enu/Examples/Visual Studio 2010_2012_2013_2015/CSharp/CustomSearchDialog/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CustomSearchDialog
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            // In the form constructor do not use any Viewer methods (Open, Print etc.) that perform UI activities.
        }

        private void pdfViewer1_SearchDialogShowing(object sender, Foxit.PDF.Viewer.SearchDialogShowingEventArgs e)
        {
            e.Cancel = true;
            searchText.Focus();
        }

        private void button3_Click(object sender, EventArgs e)
        {
            if (!String.IsNullOrEmpty(searchText.Text))
            {
                Foxit.PDF.Viewer.MatchOptions matchOptions = Foxit.PDF.Viewer.MatchOptions.All; ;
                if ((matchWholeWord.Checked == true) && (matchCase.Checked == true))
                {
                    matchOptions = Foxit.PDF.Viewer.MatchOptions.All;
                }
                else if ((matchWholeWord.Checked == true) && (matchCase.Checked == false))
                {
                    matchOptions = Foxit.PDF.Viewer.MatchOptions.WholeWordOnly;
                }
                else if ((matchWholeWord.Checked == false) && (matchCase.Checked == true))
                {
                    matchOptions = Foxit.PDF.Viewer.MatchOptions.CaseSensitive;
                }
                else
                {
                    matchOptions = Foxit.PDF.Viewer.MatchOptions.None;
                }

                pdfViewer1.SearchForward(searchText.Text, matchOptions);
            }
        }

        private void pdfViewer1_DocumentDisplayed(object sender, Foxit.PDF.Viewer.DocumentDisplayedEventArgs e)
        {
            groupBox1.Enabled = true;
            title.Text = "Type a search term in the 'Find' text box and click on 'Find Next'.";
        }

        
    }
}
